src/Parallel/Actions/SetupDataBox.hpp

```diff
@@ -0,0 +1,132 @@
+      tmpl::list<typename optional_compute_tags<ActionList>::type...>>>;
+};
+
+template <typename DbTags, typename... simple_tags, typename... compute_tags>
+static auto merge_into_databox_helper(
+    db::DataBox<DbTags>&& box, tmpl::list<simple_tags...> /*meta*/,
+    tmpl::list<compute_tags...> /*meta*/) noexcept {
+  if constexpr (not tmpl2::flat_any_v<
+                    tmpl::list_contains_v<DbTags, simple_tags>...> and
+                not tmpl2::flat_any_v<
+                    tmpl::list_contains_v<DbTags, compute_tags>...>) {
+    return db::create_from<db::RemoveTags<>, db::AddSimpleTags<simple_tags...>,
+                           db::AddComputeTags<compute_tags...>>(
+        std::move(box), typename simple_tags::type{}...);
+  } else {
+    ERROR(
+        "The SetupDataBox action must be called only once, and has either "
+        "been called more than once or one or more of the tags it is adding "
+        "are already present.");
+    return std::move(box);
+  }
+}
+}  // namespace detail
+
+/*!
+ * \brief Merge into the \ref DataBoxGroup the collection of tags requested for
+ * initialization in any of the actions.
+ *
+ * \details This action adds all of the simple tags given in the `simple_tags`
+ * type lists in each of the other actions in the current phase, and all of the
+ * compute tags given in the `compute_tags` type lists. If an action does not
+ * give either of the type lists, it is treated as an empty type list.
+ *
+ * To prevent the proliferation of many \ref DataBoxGroup types, which can
+ * drastically slow compile times, it is preferable to use only this action to
+ * add tags to the \ref DataBoxGroup, and place this action at the start of the
+ * `Initialization` phase action list. The rest of the initialization actions
+ * should specify `simple_tags` and `compute_tags`, and assign initial values to
+ * those tags, but not add those tags into the \ref DataBoxGroup.
+ */
+struct SetupDataBox {
+  template <typename DbTags, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static auto apply(db::DataBox<DbTags>& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::GlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    using action_list_simple_tags =
+        typename detail::get_action_list_simple_tags<ActionList>::type;
+    using action_list_compute_tags =
+        typename detail::get_action_list_compute_tags<ActionList>::type;
+    // grab the simple_tags, compute_tags, mutate the databox, creating
+    // default-constructed objects.
+    return std::make_tuple(detail::merge_into_databox_helper(
+        std::move(box), action_list_simple_tags{}, action_list_compute_tags{}));
```

Comment:
I occasionally encounter #2524 when merging lots of simple tags into a DataBox at once. I haven't investigated the issue in much detail, so I'm not sure if it's the number of tags or the amount of data that's relevant. If it's the number of tags, then `SetupDataBox` has a high likelihood of triggering the issue. If it's the amount of data, then the new initialization mechanism might actually fix it. Have you seen the issue at all? Any ideas what could cause it?

Reply:
I haven't encountered that error -- are there any steps you can suggest to reliably reproduce it? When you encountered it, was it locally in the container, or on a particular HPC system?

Reply:
I've also never encountered that issue... The issue is quite sparse. There are no details about compiler, env, etc. Please fill that out otherwise we can't help and should just close the issue.

Reply:
I encountered it on my laptop, trying to merge ~30 tags (maybe more) to the DataBox at once. I haven't tried reproducing it in a controlled environment yet, but that would probably worthwhile.

Reply:
Here's a patch to current `develop` that reproduces the issue on my machine: https://github.com/nilsleiffischer/spectre/commit/d59ac5b6ac1735ded441d13a3c0373c26146ca76

```
$ make -j12 ExportCoordinates3D \
    && ./bin/ExportCoordinates3D +p2 \
      --input-file tests/InputFiles/ExportCoordinates/Input3D.yaml

[1]    22117 bus error  ./bin/ExportCoordinates3D --input-file  +p2
```

Reply:
Umm 300 is _a lot_. Do you know roughly where this breaks for you?

Reply:
sorry @nilsleiffischer, I wasn't able to reproduce the problem either locally or on wheeler with your patch -- the sigbus makes me think it is related to some hardware based oddity, so this might need to be investigated by someone who has a macintosh laptop to test on.

Reply:
Thanks for checking @moxcodes. Since I also can't reproduce the problem in the Docker container we should just move on, and I'll need to track this down further. Below are some more details, but let's not hold up this PR with this discussion.

---

In this patch to `ExportCoordinates` it breaks somewhere between 200 and 300 tags when running on 2 cores, compiled on macOS with AppleClang 11 or 12. In the XCTS executable it breaks when adding a `Variables` with only 15 tags to the DataBox (the background variables such as conformal metric, energy density etc), and splitting that call into 3 separate `db::create_from`s resolves it. This makes me think it's somehow related to system load, or the size of the DataBox that is being added to. I tracked it down to this line
https://github.com/sxs-collaboration/spectre/blob/d1ac7e2676842c20ecc43fa2107122187d06d254/src/DataStructures/DataBox/DataBox.hpp#L1287
where the `new_tag_list` is a list of all tags, including the subitems from the Variables, which can be a lot. The `DataBox` constructor doesn't appear to get called, so I wonder if the issue might be in the way the DataBox allocates memory.. perhaps related to the inheritance off so many `DataBoxLeaf`s..

Reply:
this sounds like a compiler bug in AppleClang. It's not that we are inheriting too many tags. Even the Intel compiler can handle inheriting many tags. Additionally, if that were the issue it would be a problem independent of how you construct the final `DataBox`. Can you test LLVM clang on macOS? It might be that we need to drop AppleClang

Reply:
Compiling with either llvm clang 10 or 11 I get a segfault instead of the sigbus.. 🤔

Reply:
on the same line? segfaults are usually easier to diagnose than sigbus, but it's not clear that will hold here :P

Reply:
yes on the same line :(